src/components/organisms/SchedulePageModal/SchedulePageModal.tsx

```diff
@@ -1,162 +1,126 @@
+import { formatDate } from "utils/time";
 import { isEventLiveOrFuture } from "utils/event";
 
 import { useConnectRelatedVenues } from "hooks/useConnectRelatedVenues";
 import { useVenueId } from "hooks/useVenueId";
+import { useUser } from "hooks/useUser";
 
-import { EventDisplay } from "components/molecules/EventDisplay/EventDisplay";
+import { Schedule } from "components/molecules/Schedule";
+import { ScheduleVenueDescription } from "components/molecules/ScheduleVenueDescription";
 
-type DatedEvents = Array<{
-  dateDay: Date;
-  events: Array<WithVenueId<VenueEvent>>;
-}>;
+import { ScheduleDay } from "components/molecules/Schedule/Schedule.types";
+
+import { scheduleDayBuilder } from "./SchedulePageModal.utils";
+
+import "./SchedulePageModal.scss";
 
 const DAYS_AHEAD = 7;
 
 interface SchedulePageModalProps {
   isVisible?: boolean;
 }
 
+export const emptyPersonalizedSchedule = {};
+
 export const SchedulePageModal: FC<SchedulePageModalProps> = ({
   isVisible,
 }) => {
   const venueId = useVenueId();
-  const {
-    parentVenue,
-    currentVenue,
-    relatedVenues,
-    relatedVenueEvents,
-  } = useConnectRelatedVenues({
+  const { userWithId } = useUser();
+  const userEventIds =
+    userWithId?.myPersonalizedSchedule ?? emptyPersonalizedSchedule;
+
+  const { relatedVenueEvents, relatedVenues } = useConnectRelatedVenues({
     venueId,
     withEvents: true,
   });
 
-  const relatedVenuesById: Partial<
-    Record<string, WithId<AnyVenue>>
-  > = relatedVenues.reduce(itemsToObjectByIdReducer, {});
+  const [relatedRooms, setRelatedRooms] = useState<Room[]>([]);
+
+  useEffect(() => {
+    const rooms: Room[] = [];
+    relatedVenues
+      .map((venue) => venue.rooms || [])
+      .forEach((venueRooms) => rooms.push(...venueRooms));
+    setRelatedRooms(rooms);
+  }, [relatedVenues]);
 
-  const orderedEvents: DatedEvents = useMemo(() => {
+  const schedule: ScheduleDay[] = useMemo(() => {
     const liveAndFutureEvents = relatedVenueEvents.filter(isEventLiveOrFuture);
-    const hasEvents = liveAndFutureEvents.length > 0;
-
-    const nowDay = startOfDay(new Date());
-
-    const dates: DatedEvents = range(0, DAYS_AHEAD).map((idx) => {
-      const day = addDays(nowDay, idx);
-
-      const todaysEvents = liveAndFutureEvents
-        .filter((event) => {
-          return isWithinInterval(day, {
-            start: startOfDay(new Date(event.start_utc_seconds * 1000)),
-            end: endOfDay(
-              new Date(
-                (event.start_utc_seconds + event.duration_minutes * 60) * 1000
-              )
-            ),
-          });
-        })
-        .sort((a, b) => a.start_utc_seconds - b.start_utc_seconds);
-
-      return {
-        dateDay: day,
-        events: hasEvents ? todaysEvents : [],
-      };
-    });
-
-    return dates;
-  }, [relatedVenueEvents]);
-
-  const [date, setDate] = useState(0);
-
-  const scheduleTabs = useMemo(
+    const today = startOfDay(Date.now());
```

Comment:
Do we expect to use this functionality more often? Should we extract it into `utils/time`, maybe as `startOfToday` or similar?

Reply:
Actually, there is already https://date-fns.org/v2.20.1/docs/startOfToday that we could be using.